libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -72,6 +127,7 @@ internal class RouteRefreshController(
     private suspend fun refreshRoutesWithRetry(
         routes: List<NavigationRoute>
     ): RefreshedRouteInfo = coroutineScope {
+        onNewState(RouteRefreshExtra.REFRESH_STATE_STARTED)
```

Comment:
This should be moved to after `timeUntilNextAttempt.await()`. We should notify that we're starting the refresh request rounds only when we actually do, not when the routes are set.